daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -1627,6 +1627,9 @@ private[lf] final class Compiler(
       )
     case Command.LookupByKey(templateId, contractKey) =>
       LookupByKeyDefRef(templateId)(SEValue(contractKey))
+    case Command.Rollback(_) =>
+      // TODO https://github.com/digital-asset/daml/issues/8020
+      sys.error("rollback cmds are not supported")
```

Comment:
We do not want Rollback command. 
1 - Canton will not use it.
2 - The will never appear are transaction roots.

Reply:
I’m confused, you said in Slack, that you want a version of `reinterpret` that works on speedy commands. The whole point of `reinterpret` is that it works on non-root nodes. So how do you handle rollback nodes here?

Reply:
turned into a draft until I clear up my confusion

Reply:
In practice, Canton needs a `reinterpret` for the a single speedy commands as they are now: create, exercise, fetch and lookup, i.e.:
```
def reinterpret(
      submitters: Set[Party],
      action: speedy.Command,
      nodeSeed: Option[crypto.Hash],
      submissionTime: Time.Timestamp,
      ledgerEffectiveTime: Time.Timestamp,
      checkAuthorization: CheckAuthorizationMode = CheckAuthorizationMode.On,
  ): Result[(SubmittedTransaction, Tx.Metadata)]
```
Because the interface does not exists, Canton currently rebuilds a dummy node for each of those 4 case. For instance for exercising a contract, Canton build an dummy exercise node with the relevant contract ID, template ID, choice ID, but without result and with no children. Then it call the reinterpret with this dummy node, which is translate into the desired command.

By changing the interface as describe above: 
 - Canton avoids the unnecessary generation of dummy node 
 - We do not have to handle (i.e. crash) Rollback node.

Reply:
I’m still confused. Canton verifies a projection. With rollback nodes, that projection can contain a rollback node at the root. How do you call `reinterpret` on such a projection if you don’t add rollback nodes to speedy commands?

Reply:
Canton already reimplements itself the effect of several actions by using several times the reinterpret on a single node and by managing an "artificial" state of the ledger between each of those call.
Canton will have to reimplements rollback itself in a similar way.

Reply:
Alright, that seems sensible. I think we should move some of the logic that Canton currently has into the Daml repo and maybe a speedy command makes sense at that point, but let’s wait until we actually do that before adding anything.